indexer-app/src/cmr/indexer/data/index_set_generics.clj

```diff
@@ -2,36 +2,49 @@
   (:refer-clojure :exclude [update])
   (:require
    [cheshire.core :as json]
-   [clj-http.client :as client]
+   [clojure.java.io :as io]
    [clojure.string :as string]
-   [cmr.common.cache :as cache]
-   [cmr.common.concepts :as cs]
    [cmr.common.config :as cfg :refer [defconfig]]
-   [cmr.common.lifecycle :as lifecycle]
-   [cmr.common.log :as log :refer (debug info warn error)]
-   [cmr.common.services.errors :as errors]
-   [cmr.elastic-utils.index-util :as m :refer [defmapping defnestedmapping]]
+   [cmr.common.log :as log :refer (error)]
+   [cmr.elastic-utils.index-util :as m]
    [cmr.indexer.data.concepts.generic-util :as gen-util]
-   [cmr.indexer.data.index-set-elasticsearch :as index-set-es]
-   [cmr.schema-validation.json-schema :as js-validater]
-   [cmr.transmit.metadata-db :as meta-db]))
+   [cmr.schema-validation.json-schema :as js-validater]))
 
 ;; TODO: Generic work - move this to a location where index and ingest can find it
 (defn- validate-index-against-schema
   "validate a document, returns an array of errors if there are problems
    Parameters:
    * raw-json, json as a string to validate"
   [raw-json]
-  (let [schema-file (slurp (clojure.java.io/resource "schemas/index/v0.0.1/schema.json"))
+  (let [schema-file (slurp (io/resource "schemas/index/v0.0.1/schema.json"))
         schema-obj (js-validater/json-string->json-schema schema-file)]
     (js-validater/validate-json schema-obj raw-json)))
 
-;; TODO: Generic work - move this to the config file in future releases
+;; This is the default generic index number of shards.
+(def default-generic-index-num-shards 5)
+
+;; This defconfig here so that the value can be overriden by environment variable.
+;; This value can also be overriden in the schema specific configuration files. These
+;; files are found in the schemas project.
+;; Here is an example: 
+;;     "IndexSetup" : {
+;;      "index" : {"number_of_shards" : 5,
+;;                 "number_of_replicas" : 1,
+;;                 "refresh_interval" : "1s"}
+;;    },
```

Comment:
Would prefer examples be included in docstrings for the related config.

Reply:
I have delete the comment and put the contents into the docstring.